app/src/main/java/org/kontalk/util/MessageUtils.java

```diff
@@ -708,17 +735,29 @@ public static String toString(byte[] text) {
         return text;
     }
 
-    public static void convertSmileys(Editable input) {
+    public static boolean convertSmileys(Editable input) {
+        int i = 0;
         for (String key : sEmojiConverterMap.keySet()) {
-            replaceEditable(input, key, sEmojiConverterMap.get(key));
+            if (replaceEditable(input, key, sEmojiConverterMap.get(key))){
+                i++;
```

Comment:
performance and space in this regard are minimal I am more concerned with code readability. if you are set on using an int rather than a bool (why?) you could put `i=1` here instead. that is more clear that you are setting a flag and not counting something.